fdbclient/GenericManagementAPI.actor.h

```diff
@@ -918,6 +918,67 @@ Future<std::map<TenantName, TenantMapEntry>> listTenants(Reference<DB> db,
 		}
 	}
 }
+
+ACTOR template <class DB>
+Future<Void> renameTenant(Reference<DB> db, TenantName oldName, TenantName newName) {
+	state Reference<typename DB::TransactionT> tr = db->createTransaction();
+
+	state Key oldNameKey = oldName.withPrefix(tenantMapPrefix);
+	state Key newNameKey = newName.withPrefix(tenantMapPrefix);
+	state bool firstTry = true;
+	state int64_t id;
+	state Key prefix;
+	loop {
+		try {
+			tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+			state Optional<TenantMapEntry> oldEntry = wait(tryGetTenantTransaction(tr, oldName));
+			state Optional<TenantMapEntry> newEntry = wait(tryGetTenantTransaction(tr, newName));
+			if (firstTry) {
+				if (!oldEntry.present()) {
+					throw tenant_not_found();
+				}
+				if (newEntry.present()) {
+					throw tenant_already_exists();
+				}
+				// Store the id and prefix we see when first reading this key
+				id = oldEntry.get().id;
+				prefix = oldEntry.get().prefix;
+
+				firstTry = false;
+			} else {
+				// If we got commit_unknown_result, the rename may have already occurred.
+				if (!oldEntry.present() && newEntry.present()) {
+					int64_t check_id = newEntry.get().id;
```

Comment:
We generally favor camel case for variable names:

```suggestion
					int64_t checkId = newEntry.get().id;
```